tests/e2e/test_e2e.py

```diff
@@ -166,18 +165,30 @@ def test_e2e(caplog):
     Temp = mention_subclass("Temp")
 
     mention_extractor = MentionExtractor(
-        [Part, Temp], [part_ngrams, temp_ngrams], [part_matcher, temp_matcher]
+        session, [Part, Temp], [part_ngrams, temp_ngrams], [part_matcher, temp_matcher]
     )
 
     mention_extractor.apply(docs, parallelism=PARALLEL)
 
     assert session.query(Part).count() == 299
     assert session.query(Temp).count() == 134
+    assert len(mention_extractor.get_mentions()) == 2
+    assert len(mention_extractor.get_mentions()[0]) == 299
+    assert (
+        len(
+            mention_extractor.get_mentions(
+                docs=[session.query(Document).filter(Document.name == "112823").first()]
+            )[0]
+        )
+        == 70
+    )
 
     # Candidate Extraction
     PartTemp = candidate_subclass("PartTemp", [Part, Temp])
 
-    candidate_extractor = CandidateExtractor([PartTemp], throttlers=[temp_throttler])
+    candidate_extractor = CandidateExtractor(
```

Comment:
Is there a way to hide session in mention_extractor, candidate_extractor, featurizer, and labeler?

Reply:
This seems tricky. That would basically embed the session into the UDF class. I imagine it's possible, but makes the database even more coupled than it is now.